plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -121,4 +144,13 @@ private record DelegationToken(long writeTimeNanos, String delegationToken)
             requireNonNull(delegationToken, "delegationToken is null");
         }
     }
+
+    private static RuntimeException propagate(Throwable throwable)
```

Comment:
please remove this method as `Failsafe` does handle `InterruptiodException` correctly, see: `dev.failsafe.internal.RetryPolicyExecutor#apply`